src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -221,19 +222,36 @@ def specify_argument_list(self, argument_symbols):
         self._validate_arg_list(argument_symbols)
         self._argument_list = argument_symbols[:]
 
-    def lookup(self, name):
+    def lookup(self, name, scope=None):
```

Comment:
I like the filtering being done here (besides the argument name :)) because if we later redefine how the accessibility is implemented, the changes will maybe be contained inside the psyir.symbol package.